### ISSUE TITLE
Create and maintain sample mesh for face/eye tracking

### ISSUE THREAD

**Issue by gentlecolts** [Yellow-Dog-Man/Resonite-Issues#2073]
### Describe the Underlying Issue

As an avatar creator, finding information on what blend shapes to make and what they should look like for full face tracking compliance is a struggle.  What documentation does exist is often spread very thin, incomplete, and out of date, and to my knowledge there is no complete, interactive, and downloadable mesh that one could use as a reference.

What official(?) references I have been able to find are mostly 2d, non-interactive references, and use naming conventions that are inaccurate and/or out of date, and incomplete, especially now with the extra additions of quest pro shapes.

### Expected Behavior

The Resonite team should provide readily available, up to date documentation on what is expected of avatars that intend to support face tracking, including but not limited to a downloadable mesh, which can be viewed in common programs such as blender or maya, that should serve as a reference both for the visual transform of the expected shapes, and the expected naming conventions for these shapes

### Reporters

zangooseoo

**Comment by gentlecolts** [comment]
also, semi-related #1004

**Comment by shiftyscales** [comment]
Eye and Face Tracking documentation would be best hosted on the wiki instead. E.g. see the bottom of the following page: https://wiki.resonite.com/Face_and_Eye_Tracking_(Troubleshooting)

Which notes:
> Shape Keys and Setup
> Once your avatar is working please see the guide on setting up your avatar. TODO: THIS NEEDS TO BE WRITTEN. 

This could also be linked on other pages that document Resonite's avatar requirements, e.g. the humanoid rig requirements for IK: https://wiki.resonite.com/Humanoid_Rig_Requirements_for_IK

Seeking input from @Frooxius or @ProbablePrime if there's an easy to export/document list of supported eye and face blendhapes, and their detected names.

**Comment by Frooxius** [comment]
I don't think we'd commit resources to making our own face tracking mesh for this.

Particularly because individual face tracking SDK's already provide their own, which you could use as a reference. 

For the Quest Pro, you can find some FBX sample meshes here: https://github.com/oculus-samples/Unity-Movement

I used some of these during the development for testing.

**Comment by gentlecolts** [comment]
> I don't think we'd commit resources to making our own face tracking mesh for this.

As far as sample meshes, I don't really care that much if sample meshes are Resonite-made or come from elsewhere, but externally made sample meshes should definitely be linked as part of documentation.  What absolutely needs to be part of Resonite documentation is the keys themselves, I've never been able to find an exhaustive list and, as I mention in the original issue, what docs/guides I have been able to find tend to not mach up 1-to-1 with what Resonite expects/supports

**Comment by Frooxius** [comment]
The documentation is fine and that makes sense, My original comment was when the issue was still named to "Create and maintain sample mesh for face/eye tracking"

**Comment by gentlecolts** [comment]
Yeah that's all good!  I had written my original post under the assumption that there weren't really any viable mesh options out there, since I had been unable to find them.  if they exist already that's all the better, as long as Resonite has some documentation of

- what shapes are supported, what they should be named
- some links to example mesh(es) showing what those shapes should look like

This issue will be satisfied, it doesn't really matter where those meshes come from, especially since with quest pro support it seems like Resonite is combining a mix of standards

**Comment by shiftyscales** [comment]
On the "what should they look like" side of things- I found this documentation on what Quest face tracking blendshapes should look like. https://developer.oculus.com/documentation/native/android/move-ref-blendshapes/

I've seen user-made references going around in past for the SRanipal shapes, but wasn't immediately able to find any official documentation/reference visuals on what the shapes are supposed to look like otherwise.

**Comment by ProbablePrime** [comment]
I have an overhaul of our wiki content for face and eye tracking planned (at least in my head). This should also help

**Comment by zahndy** [comment]
Vive has some SRanipal blendshape info in their SDK 
[Getting Started with SRanipal in Unity Eye-v1.3.3.0.pdf](https://github.com/user-attachments/files/15899467/Getting.Started.with.SRanipal.in.Unity.Eye-v1.3.3.0.pdf)
[Getting Started with SRanipal in Unity Lip-v1.3.3.0.pdf](https://github.com/user-attachments/files/15899469/Getting.Started.with.SRanipal.in.Unity.Lip-v1.3.3.0.pdf)
The Lip doc has some issues like swapping left/right in the tongue section

**Comment by ProbablePrime** [comment]
We're beginning to cook here with: https://wiki.resonite.com/Visemes

Something I need is permissively licensed images for all of the shapes, the PDFs are great and we can link to them but i'd love to make a table in wiki and for that we need permissively licensed versions of those shapes.

**Comment by gentlecolts** [comment]
> Something I need is permissively licensed images

Is this a request for the community?  I wouldn't mind making screenshots of avatars I've made and sharing them with the community, but I'm not sure if they'd be good or representative.

Though that also brings to mind another tangentially related challenge I face as an avatar creator which is that many of my avatars have muzzles rather than flat faces and it can be challenging trying to translate that when most references are human bases, wherever reference images/models are sourced from it would be kind of nice if there were multiple types of facial structures on display alongside the familiar human bases.

**Comment by ProbablePrime** [comment]
Sure! and ideally i'd like 2 or more sets with one being a humanoid face and one being any snouted avatar's face. We could even do more!

**Comment by gentlecolts** [comment]
[Synx Face Tracking.zip](https://github.com/user-attachments/files/15921139/Synx.Face.Tracking.zip)
[viseme list.txt](https://github.com/user-attachments/files/15921147/viseme.list.txt)
Made a blender scene last night to auto render out every one of these, can make changes if required.  This was a bit lower quality than desired for time so if you think it's good I can cook out a less noisy version

Also had to use LZMA instead of Deflate for zip compression for it to fit on github, not sure if windows 100% supports but 7z should, or I can tweak the resolution/quality if that's more desirable

**Comment by ProbablePrime** [comment]
These are great!

**Comment by gentlecolts** [comment]
Cool, I'll cook up some slightly less noisy versions then, anything that could be improved?

**Comment by ProbablePrime** [comment]
Providing separate images rather than a grid might help, we'll be shoving them to the wiki

**Comment by gentlecolts** [comment]
I can do that, it's already separate frames that I just stitch together into a grid, was mostly just worried about github's file size limits

**Comment by ProbablePrime** [comment]
Feel free top shove them anywhere more suitable and link here!

**Comment by gentlecolts** [comment]
[Synx Face Tracking Combined.zip](https://github.com/user-attachments/files/15922353/Synx.Face.Tracking.Combined.zip)
[Synx Face Tracking Separate Nowire.zip](https://github.com/user-attachments/files/15922355/Synx.Face.Tracking.Separate.Nowire.zip)
[Synx Face Tracking Separate wired.zip](https://github.com/user-attachments/files/15922356/Synx.Face.Tracking.Separate.wired.zip)
Looks like it'll work, just needed to split it up

**Comment by ProbablePrime** [comment]
@gentlecolts I forgot to thank you for these, They will be helpful once I get there. Sorry that I'm a bit busy atm.
